contracts/pegged/tokens/MintSwapCanonicalToken.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity >=0.8.9;
+
+import "./MultiBridgeToken.sol";
+
+/**
+ * @title Canonical token that supports multi-bridge minter and multi-token swap
+ */
+contract MintSwapCanonicalToken is MultiBridgeToken {
+    // each bridge token.balanceOf(this) tracks how much that bridge has already swapped
+    mapping(address => uint256) public tokenSwapCap; // each bridge token -> swap cap
+
+    event TokenSwapCapUpdated(address token, uint256 cap);
+
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        uint8 decimals_
+    ) MultiBridgeToken(name_, symbol_, decimals_) {}
+
+    // msg.sender has bridge token and want to get canonical token
+    function swapBridgeForCanonical(address _bridgeToken, uint256 _amount) external {
```

Comment:
do we want to return canonical token amount?
it's not needed for this contract, but may be nice to have for future standard compatibility

Reply:
added